compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/QueryExpressionNode.java

```diff
@@ -41,13 +41,19 @@ public QueryPipelineNode queryPipeline() {
         return childInBucket(1);
     }
 
+    /**
+     * @deprecated Use {@link #resultClause()} instead.
+     */
     @Deprecated
     public SelectClauseNode selectClause() {
-        ClauseNode endClause = endClause();
-        return endClause.kind() == SyntaxKind.SELECT_CLAUSE ? (SelectClauseNode) endClause : null;
+        ClauseNode endClause = resultClause();
```

Comment:
Shouldn't `endClause` here be renamed as well?

Reply:
Yes, changed